src/pages/CourierDeliveryDetailsScreen/CourierDeliveryDetailsScreen.tsx

```diff
@@ -0,0 +1,111 @@
+import { useContext } from 'react';
+import { Form } from 'react-final-form';
+import { addDays } from 'date-fns';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import { DatePicker, TextField, TimePicker, Select } from 'mui-rff';
+import ELEVATOR_AVAILABILITY_OPTIONS from 'constants/elevatorAvailabilityOptions';
+import FormScreens from 'constants/formScreens';
+import WizardFormContext, {
+  InitialFormValuesType,
+} from 'contexts/WizardFormContext';
+import {
+  composeValidators,
+  isPositiveInteger,
+  validateMinDate,
+  validateIsRequired,
+} from 'utils/validation';
+
+export const MIN_DATE = addDays(new Date(), 3);
+
+type CourierDeliveryDetailsType =
+  InitialFormValuesType[FormScreens.COURIER_DELIVERY_DETAILS];
+
+export default function DeliveryDetailsScreen() {
+  const { formValues, onSaveFormValues } = useContext(WizardFormContext);
+  console.log(formValues);
+
+  return (
+    <Form<CourierDeliveryDetailsType>
+      onSubmit={(values) => {
+        onSaveFormValues(FormScreens.COURIER_DELIVERY_DETAILS, values);
+      }}
+      render={({ handleSubmit }) => (
+        <form onSubmit={handleSubmit}>
+          <Box sx={{ mb: 2 }}>
+            <span>Choose date and time for courier delivery</span>
+          </Box>
+          <DatePicker
+            defaultValue={MIN_DATE}
+            fieldProps={{
+              validate: composeValidators(validateIsRequired, validateMinDate),
```

Comment:
Add `parse` handler that will convert `string` to `Date` and eliminate this error